Netcracker/KubeMarine
# [CPDEV-101285] Reconfigure sysctl

### Description
* If maxPods or podPidsLimit are reconfigured in `services.kubeadm_patches.kubelet`,
  the related kernel parameter kernel.pid_max is not reconfigured.
* PaaS check fails because the parameter value is unexpected.
* The problem appeared after #639 after which calculated kernel.pid_max started to take into account kubeadm patches.

### Solution
* Support to reconfigure `services.sysctl` and `patches` using `reconfigure` procedure.
    * Added task `prepare.system.sysctl`.
    * In comparison to the installation procedure, the task does not trigger reboot.
    * `services.sysctl` is merged with the main inventory, and `patches` can only be appended.
* Implemented detecting of changes in the inventory in `prepare.system.sysctl` and `deploy.kubernetes.reconfigure` tasks.
    * If maxPods or podPidsLimit are changed for kubelet, the dependent kernel.pid_max parameter is reconfigured.
    * If net.netfilter.nf_conntrack_max parameter is changed, the dependent kube-proxy settings are reconfigured.
* Restrict the set of parameters to support reconfiguring in `services.kubeadm_patches.kubelet` to the same as for `services.kubeadm_kubelet`.
    * Support to reconfigure maxPods and podPidsLimit in both sections.

### Test Cases

**TestCase 1**

Basic kernel parameters reconfiguring scenario.

Steps:

1. Run `reconfigure` procedure providing new kernel parameters, or changing the already configured ones using `services.sysctl`, and `patches` in the procedure inventory.

ER: The parameters are reconfigured.

2. Run `check_paas`.

ER: check is successful.

**TestCase 2**

Run procedure with empty `services.sysctl` section.

Steps:

1. Change `services.sysctl` in `cluster.yaml` manually, and configure empty `services: {sysctl: {}}` section in the procedure inventory.
2. Run `reconfigure` procedure.

ER: The parameters are reconfigured.

**TestCase 3**

Change of maxPods or podPidsLimit triggers reconfiguring of sysctl.

Steps:

1. Configure new maxPods or podPidsLimit using `services.kubeadm_patches.kubelet`, or `services.kubeadm_kubelet` in the procedure inventory.
2. Run `reconfigure` procedure.

ER: Both kubelet and sysctl are reconfigured.

3. Run `check_paas`.

ER: check is successful.

**TestCase 4**

Change of net.netfilter.nf_conntrack_max triggers reconfiguring of kube-proxy.

Steps:

1. Install cluster v1.29.1.
2. Configure new `"net.netfilter.nf_conntrack_max".value` using `services.sysctl`, or `patches` in the procedure inventory.
3. Run `reconfigure` procedure.

ER: Both kube-proxy and sysctl are reconfigured.

4. Run `check_paas`.

ER: check is successful.

### Checklist

#### Unit tests
test_reconfigure.py - cover new cases.
